### PR TITLE
chore(zipkin-exporter): relay on the status code for the request but still read the response body.

### PR DESCRIPTION
**Pull request by jcchavezs**
This is a port from https://github.com/open-telemetry/opentelemetry-go/pull/1319

Ping @MrAlias

### CODE REVIEW

File: exporters/trace/zipkin/zipkin.go [hunk 1]
```diff
@@ -166,19 +167,21 @@ func (e *Exporter) ExportSpans(ctx context.Context, batch []*export.SpanData) er
...
 	if err != nil {
-		// Best effort to clean up here.
-		resp.Body.Close()
 		return e.errf("failed to read response body: %v", err)
 	}
 
-	err = resp.Body.Close()
-	if err != nil {
-		return e.errf("failed to close response body: %v", err)
+	if resp.StatusCode != 202 {
```
Comment (cbandy) [inline exporters/trace/zipkin/zipkin.go:181]:
```suggestion
	if resp.StatusCode != http.StatusAccepted {
```

File: exporters/trace/zipkin/zipkin.go [hunk 2]
```diff
@@ -166,19 +167,21 @@ func (e *Exporter) ExportSpans(ctx context.Context, batch []*export.SpanData) er
...
 	}
-	e.logf("zipkin responded with status %d", resp.StatusCode)
+	defer resp.Body.Close()
 
-	_, err = ioutil.ReadAll(resp.Body)
+	// Zipkin API returns a 202 on success and the content of the body isn't interesting
+	// but it is still being read because according to https://golang.org/pkg/net/http/#Response
+	// > The default HTTP client's Transport may not reuse HTTP/1.x "keep-alive" TCP connections
+	// > if the Body is not read to completion and closed.
+	_, err := io.Copy(ioutil.Discard, resp.Body)
```
Comment (cbandy) [inline exporters/trace/zipkin/zipkin.go:176]: 👍 iiuc, this hits the `ReadFrom` fast path.

**Comment by jcchavezs** [comment]
Can we merge this @MrAlias?

**Comment by Aneurysm9** [comment]
> Can we merge this @MrAlias?

This is not yet ready for merging.  It does not have two approvals, it does not have a changelog entry or the `Skip Changelog` label, the CI build is failing due to a compile error, and it is not able to be applied to the trunk as a fast-forward.

**Comment by jcchavezs** [comment]
> This is not yet ready for merging. It does not have two approvals, it does not have a changelog entry or the Skip Changelog label, the CI build is failing due to a compile error, and it is not able to be applied to the trunk as a fast-forward.

My bad, I misinterpreted the CI result and I thought it was only missing the changelog label. Fixing it now.

File: exporters/trace/zipkin/zipkin_test.go [hunk 3]
```diff
@@ -340,17 +341,15 @@ func TestExportSpans(t *testing.T) {
 	ctx := context.Background()
 	require.Len(t, ls.Messages, 0)
 	require.NoError(t, exporter.ExportSpans(ctx, spans[0:1]))
-	require.Len(t, ls.Messages, 2)
+	require.Len(t, ls.Messages, 1)
 	require.Contains(t, ls.Messages[0], "send a POST request")
-	require.Contains(t, ls.Messages[1], "zipkin responded")
```
Comment (jcchavezs) [inline exporters/trace/zipkin/zipkin_test.go:345]: This message is now an error if not 202 hence no need for this log anymore.

File: exporters/trace/zipkin/zipkin.go (hunk 1 above)
Comment (MrAlias) [reply exporters/trace/zipkin/zipkin.go:181]: This seems like a good suggestion. Can we include this change?

File: exporters/trace/zipkin/zipkin.go [hunk 4]
```diff
@@ -166,19 +167,21 @@ func (e *Exporter) ExportSpans(ctx context.Context, batch []*export.SpanData) er
...
 		return e.errf("request to %s failed: %v", e.url, err)
 	}
-	e.logf("zipkin responded with status %d", resp.StatusCode)
+	defer resp.Body.Close()
 
-	_, err = ioutil.ReadAll(resp.Body)
+	// Zipkin API returns a 202 on success and the content of the body isn't interesting
+	// but it is still being read because according to https://golang.org/pkg/net/http/#Response
+	// > The default HTTP client's Transport may not reuse HTTP/1.x "keep-alive" TCP connections
+	// > if the Body is not read to completion and closed.
```
Comment (MrAlias) [inline exporters/trace/zipkin/zipkin.go:175]: :+1:

**Review by MrAlias — APPROVED** [review]
LGTM overall.

File: exporters/trace/zipkin/zipkin_test.go [hunk 5]
```diff
@@ -192,6 +192,7 @@ func (c *mockZipkinCollector) handler(w http.ResponseWriter, r *http.Request) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	c.models = append(c.models, models...)
+	w.WriteHeader(202)
```
Comment (Aneurysm9) [inline exporters/trace/zipkin/zipkin_test.go:195]:
```suggestion
	w.WriteHeader(http.StatusAccepted)
```

**Review by jmacd — APPROVED** [review]

**Review by XSAM — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Comment by jcchavezs** [comment]
Can we merge it now?

**Comment by XSAM** [comment]
> Shall we merge it?

Please add a CHANGELOG entry. :)

File: exporters/trace/zipkin/zipkin.go (hunk 1 above)
Comment (XSAM) [reply exporters/trace/zipkin/zipkin.go:181]: 👍

**Comment by jcchavezs** [comment]
@XSAM done

**Comment by XSAM** [comment]
@jcchavezs There are some comments that need to solve. It looks like forced push overwrites these changes.

e.g. 202 -> http.StatusAccepted

**Comment by jcchavezs** [comment]
@XSAM yeah I just saw it. Fixed them.

**Review by XSAM — APPROVED** [review]
LGTM 

@Aneurysm9 @MrAlias I believe we should merge this PR.

**Review by dashpole — APPROVED** [review]

**Comment by jcchavezs** [comment]
Shall we?

**Comment by MrAlias** [comment]
Thanks for the contribution @jcchavezs! Sorry about the delay getting it merged.

**Comment by jcchavezs** [comment]
Thanks.

**Pull request merged**